runhouse/resources/envs/utils.py

```diff
@@ -49,12 +49,27 @@ def _get_env_from(env):
         return Env(reqs=env, working_dir="./", name=Env.DEFAULT_NAME)
     elif isinstance(env, Dict):
         return Env.from_config(env)
-    elif (
-        isinstance(env, str)
-        and Env.DEFAULT_NAME not in env
-        and rns_client.exists(env, resource_type="env")
-    ):
-        return Env.from_name(env)
+    elif isinstance(env, str):
+        if rns_client.exists(env, resource_type="env"):
+            return Env.from_name(env)
+        if system:
+            if system.get(env):
+                return system.get(env)
+
+            # handle case where env is passed in as full rns address
+            name, _ = rns_client.split_rns_name_and_path(
+                rns_client.resolve_rns_path(env)
+            )
+            if system.get(name):
+                return system.get(name)
+
+        if env == Env.DEFAULT_NAME:
+            return Env(name=Env.DEFAULT_NAME)
+
+    warning = f"Could not locate Env from value: {env}"
+    warning = f"{warning} on cluster {system.name}" if system else warning
+    logging.warning(warning)
```

Comment:
@dongreenberg can you double check my logic here? and is there an instance where it still makes sense to return the input `env` if it doesn't satisfy any of the conditions or to simply throw an error?

most cases we expect an `Env` to be returned which will throw an error elsewhere if it returns a string and we try to call an env property on it. but there could also be cases where the output is simply used to populate a config etc, not sure if it's an issue if a string is returned if the corresponding Env could not be found.

Reply:
tbh I'm nervous that passing system here is a little non-deterministic. Den and this system could have different configs for that env name, and the only thing determining which is returned is the order we've implemented here. What case are we solving for where we want to check on the cluster for the env config?

I also don't think I'd raise a warning here. We use this method all the time in instances where we don't necessarily expect and env to be found (but should construct one if it is), and it'll be pretty noisy for the user.